cesium/internal/unary/writer.go

```diff
@@ -249,7 +266,7 @@ func (w *Writer) commitWithEnd(ctx context.Context, end telem.TimeStamp) (telem.
 			return 0, err
 		}
 		if !approx.Exact() {
-			return 0, errors.New("could not get exact timestamp")
+			return 0, errors.Wrapf(validate.Error, "writer start %s cannot be resolved in the index", w.Start)
```

Comment:
Would be good to include the name of the index channel here. I'm fine with adding a public method to the index to return information about its name/key for debugging.

Reply:
This condition should be inconceivable, but I added a public method `Info` to index still – we will need to log similar errors in handling deletion.